### PR TITLE
PLG: new Mixtral 8x22B Instruct model for chat

### PR DESCRIPTION
**Pull request by abeatrix**
Depends https://github.com/sourcegraph/cody/pull/3790 -  do not merge until https://github.com/sourcegraph/cody/pull/3790  is deployed

### Status: On Hold - testing

This pull request adds a new Mixtral 8x22B Instruct chat model to the list of available models for PLG users. 
The new model is marked as Cody Pro only and has the same maximum token limit as the existing Mixtral 8x7B model.

- Added a new model entry for 'Mixtral 8x22B Preview' with the appropriate model ID, provider, and usage details.

This change provides Cody Pro users with access to the new Mixtral 8x22B model added in https://github.com/sourcegraph/sourcegraph/pull/61763 once it is deployed.

## Test plan

1. See if the new model shows up in your dropdown list. 
2. Verify the response returned by the model 

image

### CODE REVIEW

**Review by philipp-spiess — APPROVED** [review]
LGTM but let's test this end to end before we merge this PR. I’m concerned that it wont do well in chat without instruction tuning. 

@abeatrix is it fine if I roll out the dotcom change and the cody gateway change tomorrow my am? it's a bit late now and the sg/sg build will take another half hour or so first :sigh:

**Comment by abeatrix** [comment]
> LGTM but let's test this end to end before we merge this PR. I’m concerned that it wont do well in chat without instruction tuning.
> 
> @abeatrix is it fine if I roll out the dotcom change and the cody gateway change tomorrow my am? it's a bit late now and the sg/sg build will take another half hour or so first :sigh:

YEa won't merge it until we have confirmed the new model is ready 😄

File: lib/shared/src/models/dotcom.ts [hunk 1]
```diff
@@ -76,13 +76,22 @@ export const DEFAULT_DOT_COM_MODELS: ModelProvider[] = [
...
+        maxToken: DEFAULT_CHAT_MODEL_TOKEN_LIMIT,
+    },
+    {
+        title: 'Mixtral 8x22B Preview',
+        model: 'fireworks/accounts/fireworks/models/mixtral-8x22b-instruct-preview',
+        provider: 'Mistral',
+        default: false,
+        codyProOnly: true,
         usage: [ModelUsage.Chat],
         maxToken: DEFAULT_CHAT_MODEL_TOKEN_LIMIT,
```
Comment (philipp-spiess) [inline lib/shared/src/models/dotcom.ts:96]: The model only has a 2k context window, we should change this here: https://fireworks.ai/models/fireworks/mixtral-8x22b-instruct-preview

**Review by philipp-spiess — APPROVED** [review]

File: lib/shared/src/models/dotcom.ts (hunk 1 above)
Comment (abeatrix) [reply lib/shared/src/models/dotcom.ts]: Thanks for pointing that out. I've set it to 1.8k to leave some room for errors 😄

**Pull request merged**